apis/Google.Cloud.Logging.Console/Google.Cloud.Logging.Console.Tests/GoogleCloudConsoleFormatterAugmentationTest.cs

```diff
@@ -0,0 +1,109 @@
+    {
+        /// <summary>
+        /// Augment with a simple key-value pair.
+        /// </summary>
+        public void AugmentLog<T>(LogEntry<T> logEntry, Utf8JsonWriter writer)
+        {
+            writer.WriteString("simple_key", "simple_value");
+        }
+    }
+    private readonly GoogleCloudConsoleFormatterOptions simpleOptions = new GoogleCloudConsoleFormatterOptions { LogAugmenter = new SimpleAugmenter() };
+
+    [Fact]
+    public void SimplestLog()
+    {
+        var actualJson = LogSimpleLogEntry(simpleOptions);
+        Assert.Equal(ExpectedAugmentedLogEntryJson, actualJson);
+    }
+
+    private class ErrorAnnotationAugmenter : IGoogleCloudConsoleLogAugmenter
+    {
+        private static readonly JsonEncodedText s_jsonPayloadTypePropertyName = JsonEncodedText.Encode("@type");
+        private static readonly JsonEncodedText s_jsonPayloadTypePropertyValue = JsonEncodedText.Encode("type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent");
+        public void AugmentLog<T>(LogEntry<T> logEntry, Utf8JsonWriter writer)
+        {
+            if (logEntry.LogLevel >= LogLevel.Error)
+            {
+                writer.WriteString(s_jsonPayloadTypePropertyName, s_jsonPayloadTypePropertyValue);
+            }
+        }
+    }
+    private readonly GoogleCloudConsoleFormatterOptions errorAnnotationOptions = new GoogleCloudConsoleFormatterOptions { LogAugmenter = new ErrorAnnotationAugmenter() };
+
+    [Theory]
+    [InlineData(LogLevel.Trace)]
+    [InlineData(LogLevel.Debug)]
+    [InlineData(LogLevel.Information)]
+    [InlineData(LogLevel.Warning)]
+    public void NonErrorLogTest(LogLevel logLevel)
+    {
+        LogEntry<string> logEntry = new LogEntry<string>(logLevel, "LogCategory", new EventId(1), "Not so bad", exception: null, (state, exception) => state);
+        var actualJson = LogLogEntry(logEntry, errorAnnotationOptions);
+        Assert.DoesNotContain("@type", actualJson);
+        Assert.DoesNotContain("type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent", actualJson);
+    }
+
+    [Theory]
+    [InlineData(LogLevel.Error)]
+    [InlineData(LogLevel.Critical)]
+    public void ErrorLogTest(LogLevel logLevel)
+    {
+        LogEntry<string> logEntry = new LogEntry<string>(logLevel, "LogCategory", new EventId(1), "Not so bad", exception: null, (state, exception) => state);
+        var actualJson = LogLogEntry(logEntry, errorAnnotationOptions);
+        Assert.Contains("\"@type\":\"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent\"", actualJson);
+    }
+
+
+    private readonly GoogleCloudConsoleFormatterOptions NullAnnotationOptions = new GoogleCloudConsoleFormatterOptions { LogAugmenter = null };
+
+    [Fact]
+    public void NullAugmentLogTest()
+    {
+        var actualJson = LogSimpleLogEntry(NullAnnotationOptions);
+        Assert.Equal(ExpectedSimpleLogEntryJson, actualJson);
+    }
```

Comment:
I don't think we need this, it's the default and it's already being tested in GoogleCloudConsoleFormatterTest.SimplestLog.

Maybe you want to test the options type itself to check that the default value for the augmenter is null.